twill-yarn/src/main/java/org/apache/twill/internal/appmaster/RunningContainers.java

```diff
@@ -113,9 +117,11 @@ public Integer apply(BitSet input) {
   private final Location applicationLocation;
   private final Set<String> runnableNames;
   private final Map<String, Map<String, String>> logLevels;
+  private final Map<String, Integer> maxRetries;
```

Comment:
Instance ids for runnable instances go from `0` to `num-instances - 1`. If an instance gets restarted, then it retains the same instance id. Using this instance id we can track the number of times a particular instance was restarted.

You can get the instance id in `handleCompleted` method by using `getInstanceId(RunId runId)` method.